lib/src/config/dictionaries.rs

```diff
@@ -0,0 +1,143 @@
+use {
+    crate::wiggle_abi::types::DictionaryHandle,
+    std::{
+        collections::HashMap,
+        sync::{atomic::AtomicU32, Arc},
+    },
+};
+
+static COUNTER: AtomicU32 = AtomicU32::new(1);
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct DictionaryName(pub String);
```

Comment:
I've removed the internal pub and implemented `Display` and `Hash` -- `Display` is required because we of the errors and tracing in [`lib/src/config/dictionaries.rs`](https://github.com/fastly/Viceroy/pull/61/files#diff-07a2e6099d701192a7ccd4206311fc29081f2b6d753b4427286f450b3b874b3eR133-R137)